pkg/gitparse/gitparse.go

```diff
@@ -42,6 +34,46 @@ type Diff struct {
 	IsBinary  bool
 }
 
+// Parser sets values used in GitParse.
+type Parser struct {
+	MaxDiffSize   int
```

Comment:
nit: I think we can make all the fields in here private to reduce the exposure? If we can then we would just update the reaming functions to account for this change.